### DISCUSSION TITLE
Why Do We Need to Use NetworkX Nx.Graph?

### DISCUSSION THREAD

**Discussion by SpikeYangRc** [microsoft/graphrag#1409] [Q&A]
Hi, 
Based on code analysis, I noticed that although the system declares node_embedding_algorithm=node2vec as the embedding algorithm, it doesn't actually utilize the _node2vec_embed function for embedding computation. When retrieving entities, relationships, and text units, the system only uses basic graph operations like get_node and get_edge to query direct relationships - functionality that could be easily implemented using a traditional relational database. So, is the main purpose of using a graph database simply for data storage and degree calculation for ranking? Notably, the current implementation doesn't leverage the graph database's strengths in handling multi-level indirect relationships, such as discovering indirect paths from A to D through intermediate nodes B and C (A -> B -> C -> D).

**Comment by tarnished292** [comment]
w

**Comment by tarnished292** [comment]
w
